compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -851,40 +902,387 @@ installDepsAndInitPackageDb opts (InitPkgDb shouldInit) =
+
+          pure valid
+        ) sourceDepPids
+
+      log $ "Source dependencies are " <> (if sourceDepsCorrect then "not " else "") <> "stale."
+      log $ "Source files are " <> (if sourceFilesCorrect then "not " else "") <> "stale."
+      pure $ if sourceDepsCorrect && sourceFilesCorrect then getArchiveDalfPid bmName bmVersion else Nothing
+
+buildMultiRule
+  :: AssistantRunner
+  -> BuildableDataDeps
+  -> MultiPackageNoCache
+  -> Maybe (NormalizedFilePath, IO LF.PackageId)
+  -> Rules ()
+buildMultiRule assistantRunner buildableDataDeps (MultiPackageNoCache noCache) mRootPackage =
+  defineEarlyCutoffWithDefaultRunChanged $ \BuildMulti path -> do
+    logger <- actionLogger
+
+    liftIO $ IDELogger.logDebug logger $ T.pack $ "Considering building: " <> fromNormalizedFilePath path
+
+    let filePath = fromNormalizedFilePath path
+    bmPkgConfig@BuildMultiPackageConfig {..} <- liftIO $ buildMultiPackageConfigFromDamlYaml filePath
+
+    -- Wrap up a PackageId and changed flag into the expected Shake return structure
+    let makeReturn :: LF.PackageId -> Bool -> (Maybe B.ByteString, RunChanged, IdeResult (LF.PackageName, LF.PackageVersion, LF.PackageId))
+        makeReturn pid changed = 
+          ( Just $ encodeUtf8 $ LF.unPackageId pid
+          , if changed then ChangedRecomputeDiff else ChangedRecomputeSame
+          , ([], Just (bmName, bmVersion, pid))
+          )
+
+    toBuild <- liftIO $ withCurrentDirectory filePath $ flip mapMaybeM bmDataDeps $ \darPath -> do
+      canonDarPath <- canonicalizePath darPath
+      pure $ getDataDepSource buildableDataDeps canonDarPath
+
+    -- Build all our deps!
+    sourceDepsData <- uses_ BuildMulti $ toNormalizedFilePath' <$> toBuild
+
+    liftIO $ case mRootPackage of
+      -- We're building the root package, we'll envoke the normal builder directly with the Options already applied
+      Just (pkgRootPath, builder) | path == pkgRootPath -> do
+        IDELogger.logInfo logger $ T.pack $ "Building " <> filePath
+        pkgId <- builder
+        pure $ makeReturn pkgId True
+      _ -> do
+        -- Check our own staleness, if we're not stale, give the package ID to return to shake.
+        -- If caching disabled, we fail this check pre-emtively
+        ownValidPid <- 
+          if noCache
+            then pure Nothing
+            else getValidPackageId logger filePath bmPkgConfig sourceDepsData
+
+        case ownValidPid of
+          Just pid -> do
+            IDELogger.logInfo logger $ T.pack $ filePath <> " is unchanged."
+            pure $ makeReturn pid False
+          Nothing -> do
+            IDELogger.logInfo logger $ T.pack $ "Building " <> filePath
+            -- Invoke damlc with the following flags. For now we call to `main`, but later we'll defer to the assistant to find the right version of damlc
+            -- and defer to that instead.
```

Comment:
Is this comment still correct? Doesn't the daml assistant get invoked now?

Reply:
Ah oops, thats a relic